migration/src/main/java/io/apiman/migration/fix_pre_21/ApimanExportVisitor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Scheer PAS Schweiz AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apiman.migration.fix_pre_21;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Simplified visitor for iterating Org -> Api -> API Version
+ *
+ * @author Marc Savy {@literal <marc@blackparrotlabs.io>}
+ */
+public abstract class ApimanExportVisitor {
+    private static final Logger LOGGER = LogManager.getLogger(ApimanExportVisitor.class);
+
+    public ApimanExportVisitor(JsonNode root) throws Exception {
+        parse(root);
+    }
+
+    public void parse(JsonNode root) throws Exception {
+        ArrayNode orgArray = getOrThrowArray(root, "Orgs");
+
+        String orgId;
+        String apiId;
+
+        for (JsonNode orgNode : orgArray) {
+            orgId = getOrThrow(orgNode, "OrganizationBean.id").asText();
+            ArrayNode apiArray = getOrThrowArray(orgNode, "Apis");
+
+            LOGGER.trace("Visiting Org: {}", orgId);
+            visitOrg(orgId, orgNode);
+
+            for (JsonNode apiNode : apiArray) {
+                apiId = getOrThrow(apiNode, "ApiBean.id").asText();
+                ArrayNode versionArray = getOrThrowArray(apiNode, "Versions");
+
+                LOGGER.trace("Visiting API: {}", apiId);
+                visitApi(apiId, apiNode);
+
+                for (JsonNode versionNode : versionArray) {
+                    LOGGER.trace("Visiting API Version: {}", versionNode);
+                    visitApiVersion(orgId, apiId, versionNode);
+                }
+            }
+        }
+    }
+
+    public void visitOrg(String orgId, JsonNode orgNode) throws Exception {
```

Comment:
Whats the idea behind that?

Reply:
Idea behind what, sorry?

Reply:
Ahh sorry, was not clear enough.
I meant the empty void methods. Just preparation for the future or something like that?

Reply:
The idea is that the person wanting to use this visitor can override those empty methods if they want in order to visit that particular entity. 

But it's optional, so they are not labelled as abstract methods but instead are just empty.

You can then do something like 

```java
new ApimanExportVisitor() {

  <snip>

  public void visitOrg(String orgId, JsonNode orgNode) throws Exception {
	System.out.println("Do something with " + orgId);
  }
};
```

Reply:
Okay, thank you for this example :) Makes sense now!